protocol/common/message_processor.go

```diff
@@ -262,7 +264,19 @@ func (p *MessageProcessor) sendPrivate(
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to send message with datasync")
 		}
-		rawMessage.DataSyncID = datasyncID
+		// We don't need to receive confirmations from our own devices
+		if !IsPubKeyEqual(recipient, &p.identity.PublicKey) {
+			confirmation := &RawMessageConfirmation{
+				DataSyncID: datasyncID,
+				MessageID:  messageID,
+				PublicKey:  crypto.CompressPubkey(recipient),
+			}
+
+			err = p.persistence.InsertConfirmation(confirmation)
```

Comment:
I'd suggest to rename this function because it gives false feeling that we already got confirmation. 
Maybe it can be named like `InsertAwaitedConfirmation()`

To be honest, I think that probably even `raw_message_confirmations` isn't ideal naming for database table, because what we really store - is info about datasync messages - their ids, confirmation times, correspondence to messageids. And info about confirmations - is what we deduct from it using additional external info (`atLestOne` flag). But I don't have a good suggestion on how to change this, so not going to be pesky about 😁